idealingua-v1/idealingua-v1-runtime-rpc-http4s/src/main/scala/izumi/idealingua/runtime/rpc/http4s/HttpServer.scala

```diff
@@ -67,78 +66,71 @@ class HttpServer[C <: Http4sContext]
       val methodId = IRTMethodId(IRTServiceId(service), IRTMethodName(method))
       run(new HttpRequestContext(request, ctx), body = "{}", methodId)
 
-    case request@POST -> Root / service / method as ctx =>
+    case request @ POST -> Root / service / method as ctx =>
       val methodId = IRTMethodId(IRTServiceId(service), IRTMethodName(method))
       request.req.decode[String] {
         body =>
           run(new HttpRequestContext(request, ctx), body, methodId)
       }
   }
 
-  protected def handleWsClose(context: WebsocketClientContext[C#BiIO, C#ClientId, C#RequestContext]): Unit = {
-    logger.debug(s"${context -> null}: Websocket client disconnected")
+  protected def handleWsClose(context: WebsocketClientContext[C#BiIO, C#ClientId, C#RequestContext]): MonoIO[Unit] = {
+    F.sync(logger.debug(s"${context -> null}: Websocket client disconnected")) *>
     context.finish()
   }
 
-  protected def onWsOpened(): Unit = {
-  }
+  protected def onWsOpened(): MonoIO[Unit] = F.unit
 
-  protected def onWsUpdate(maybeNewId: Option[C#ClientId], old: WsClientId[ClientId]): Unit = {
+  protected def onWsUpdate(maybeNewId: Option[C#ClientId], old: WsClientId[ClientId]): MonoIO[Unit] = F.sync {
     (maybeNewId, old).forget
   }
 
-  protected def onWsClosed(): Unit = {
-  }
+  protected def onWsClosed(): MonoIO[Unit] = F.unit
 
   protected def setupWs(request: AuthedRequest[MonoIO, RequestContext], initialContext: RequestContext): MonoIO[Response[MonoIO]] = {
     val context = new WebsocketClientContextImpl[C](c, request, initialContext, listeners, wsSessionStorage, logger) {
-
-      override def onWsSessionOpened(): Unit = {
-          onWsOpened()
-          super.onWsSessionOpened()
+      override def onWsSessionOpened(): C#MonoIO[Unit] = {
+        onWsOpened() *> super.onWsSessionOpened()
       }
-
-      override def onWsClientIdUpdate(maybeNewId: Option[C#ClientId], oldId: WsClientId[C#ClientId]): Unit = {
-        onWsUpdate(maybeNewId, oldId)
-        super.onWsClientIdUpdate(maybeNewId, oldId)
+      override def onWsClientIdUpdate(maybeNewId: Option[C#ClientId], oldId: WsClientId[C#ClientId]): C#MonoIO[Unit] = {
+        onWsUpdate(maybeNewId, oldId) *> super.onWsClientIdUpdate(maybeNewId, oldId)
       }
-
-      override def onWsSessionClosed(): Unit = {
-        onWsClosed()
-        super.onWsSessionClosed()
+      override def onWsSessionClosed(): C#MonoIO[Unit] = {
+        onWsClosed() *> super.onWsSessionClosed()
       }
     }
-    context.start()
-    logger.debug(s"${context -> null}: Websocket client connected")
-
-    context.queue.flatMap[Throwable, Response[MonoIO]] { q =>
-      val dequeueStream = q.dequeue.through {
-        stream =>
-          stream
-            .evalMap(handleWsMessage(context))
-            .collect { case Some(v) => WebSocketFrame.Text(v) }
+    for {
+      _ <- context.start()
+      _ <- F.sync(logger.debug(s"${context -> null}: Websocket client connected"))
```

Comment:
@Caparow How about using `LogBIO` instead of manually wrapping these logger calls each time?